### PR TITLE
refactor(UtilService): Move sortByServerSaveTime()

### LINKED ISSUE #1160: refactor(UtilService): Move sortByServerSaveTime()
This function doesn't need to be in UtilService. Move to its own file and update references.

### PR DESCRIPTION
**Pull request by breity**
## Changes
- Moved `sortByServerSaveTime()` to `object.ts` and renamed function to better describe purpose as a comparator.
- Replaced references and removed unused UtilService dependency.

## Test
- Make sure Grade by Step loads and functions as before and that student work revisions are sorted properly (new -> old).
- Make sure Discussion posts load and display in the proper order (new -> old).
- Make sure deleted Discussion posts are still hidden from student view.

Closes #1160.

### CODE REVIEW

File: src/assets/wise5/common/object/object.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export function copy(obj: any): any {
   return typeof obj === 'undefined' ? undefined : JSON.parse(JSON.stringify(obj));
 }
+
+export function serverSaveTimeComparator(object1: any, object2: any): number {
+  return object1.serverSaveTime - object2.serverSaveTime;
+}
```
Comment (hirokiterashima) [inline src/assets/wise5/common/object/object.ts]: I think adding type to the input object params would make this function input more clear.
```suggestion
export function serverSaveTimeComparator(
  object1: { serverSaveTime: number },
  object2: { serverSaveTime: number }
): number {
  return object1.serverSaveTime - object2.serverSaveTime;
}
```

File: src/assets/wise5/components/discussion/discussionService.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { forkJoin, Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
+import { serverSaveTimeComparator } from '../../common/object/object';
 
 @Injectable()
 export class DiscussionService extends ComponentService {
```
Comment (hirokiterashima) [inline src/assets/wise5/components/discussion/discussionService.ts:10]: Remove UtilService from import and constructor?

**Review by hirokiterashima — APPROVED** [review]
Looks good. Can you address minor comments inline before merging?

File: src/assets/wise5/common/object/object.ts (hunk 1 above)
Comment (breity) [reply src/assets/wise5/common/object/object.ts]: Yep, makes sense.

**Pull request merged**